tensorflow/training/docker/2.12/py3/Dockerfile.cpu

```diff
@@ -159,7 +156,15 @@ RUN ${PIP} install --no-cache-dir -U \
     werkzeug \
     psutil \
     urllib3 \
-    "protobuf<4"
+    "protobuf<4" \
+    wheel
+
+# Use --no-build-isolation to force pyyaml 5.4.1 to be installed using Cython<3 to avoid installation errors
+# Force the installation of pyyaml<6 to prevent conflicts with tf-models-official
+RUN ${PIP} install --no-cache-dir --no-build-isolation \
+   "Cython<3" \
+   "pyYAML>=5.4.1,<6" \
+   "awscli<2"
```

Comment:
Question. as latest awscli<2 does use pyyaml>=6, wondering if the awscli version is being held back or is PyYaml is being updated to v6.x after this is completed.

Reply:
`awscli<2` is what we install by default on images, because `awscli` on pypi defaults to the latest `1.x` version. This `<2` version constraint is just an additional precaution, which existed in another dockerfile from which I copied the Cython bug-fix.

Though yes, Pyyaml will need to be updated after this is completed.